Change Boost.Python/numpy target from 3.7 to 3.8.

Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2021/12
diff -ru eman2-2.91/cmake/Boost.cmake eman2-2.91_switch_boost_target_python38/cmake/Boost.cmake
--- eman2-2.91/cmake/Boost.cmake	2021-03-08 18:13:26.000000000 +0100
+++ eman2-2.91_switch_boost_target_python38/cmake/Boost.cmake	2021-12-22 18:39:16.224016158 +0100
@@ -1,12 +1,12 @@
 set(Boost_USE_MULTITHREADED ON)
 set(Boost_NO_BOOST_CMAKE ON)
-find_package(Boost COMPONENTS python37 numpy37 REQUIRED)
+find_package(Boost COMPONENTS python38 numpy38 REQUIRED)
 
 message("Boost_LIBRARIES:   ${Boost_LIBRARIES}")
 message("Boost_INCLUDE_DIR: ${Boost_INCLUDE_DIR}")
 
 #this definition is for boost.python > 1.35.0 
-set_target_properties(Boost::python37
+set_target_properties(Boost::python38
 					  PROPERTIES
 					  INTERFACE_COMPILE_DEFINITIONS BOOST_PYTHON_NO_PY_SIGNATURES
 					  INTERFACE_LINK_LIBRARIES Python::Python
Only in eman2-2.91_switch_boost_target_python38/cmake: .Boost.cmake.swp
Only in eman2-2.91_switch_boost_target_python38: .CMakeLists.txt.swp
diff -ru eman2-2.91/libpyEM/CMakeLists.txt eman2-2.91_switch_boost_target_python38/libpyEM/CMakeLists.txt
--- eman2-2.91/libpyEM/CMakeLists.txt	2021-03-08 18:13:26.000000000 +0100
+++ eman2-2.91_switch_boost_target_python38/libpyEM/CMakeLists.txt	2021-12-22 18:57:08.398586868 +0100
@@ -11,7 +11,7 @@
 			PROPERTIES
 			PREFIX lib
 			SUFFIX ${suffix}
-			LINK_LIBRARIES "EM2;Boost::python37"
+			LINK_LIBRARIES "EM2;Boost::python38"
 			)
 	
 	install(TARGETS ${libname} DESTINATION ${SP_DIR})
@@ -66,5 +66,5 @@
 
 add_subdirectory(qtgui)
 
-target_link_libraries(pyUtils2         NumPy Boost::numpy37)
-target_link_libraries(pyTypeConverter2 NumPy Boost::numpy37)
+target_link_libraries(pyUtils2         NumPy Boost::numpy38)
+target_link_libraries(pyTypeConverter2 NumPy Boost::numpy38)
